src/Management/test/Endpoint.Test/Actuators/CloudFoundry/CloudFoundrySecurityMiddlewareTest.cs

```diff
@@ -353,18 +325,88 @@ public async Task Redacts_HTTP_headers()
         builder.Services.AddLogging(options => options.SetMinimumLevel(LogLevel.Trace).AddProvider(capturingLoggerProvider));
         builder.Services.AddCloudFoundryActuator();
         await using WebApplication app = builder.Build();
+        app.Services.GetRequiredService<HttpClientHandlerFactory>().Using(CloudControllerPermissionsMock.GetHttpMessageHandler());
         await app.StartAsync(TestContext.Current.CancellationToken);
 
         using HttpClient httpClient = app.GetTestClient();
         var requestMessage = new HttpRequestMessage(HttpMethod.Get, new Uri("http://localhost/cloudfoundryapplication"));
-        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", "some");
+        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", MockAccessToken);
 
         _ = await httpClient.SendAsync(requestMessage, TestContext.Current.CancellationToken);
 
         string logMessages = string.Join(System.Environment.NewLine, capturingLoggerProvider.GetAll());
         logMessages.Should().Contain("Authorization: *");
     }
 
+    [ClassData(typeof(CloudFoundrySecurityMiddlewareTestScenarios))]
+    [Theory]
+    public async Task IntegrationTestReturnsExpected(string scenario, HttpStatusCode? steeltoeStatusCode, string? errorMessage, string[] expectedLogs,
```

Comment:
In case of a timeout, _two_ messages are logged at Error level. There should be only one at that level:

![image](https://github.com/user-attachments/assets/2f8bf51f-44bd-49da-af03-87d0e709212e)

Reply:
Oh right... I had a change pending in the middleware but scrapped it at some point. OK with just making the permissions provider a warning?